src/components/modebar/buttons.js

```diff
@@ -513,6 +513,17 @@ function toggleHover(gd) {
     Registry.call('_guiRelayout', gd, 'hovermode', newHover);
 }
 
+modeBarButtons.resetSankeyGroup = {
+    name: 'resetSankeyGroup',
+    title: function(gd) { return _(gd, 'Ungroup all nodes'); },
+    icon: Icons.home,
+    click: function(gd) {
+        Registry.call('restyle', gd, {
+            'node.groups': [[]],
```

Comment:
As we discussed in person, it might be better to `restyle` back to the "initial" view rather than removing all groups.

To do so:

- stash the 'initial' `node.(x|y)` and `groups` in `fullData[i]` (see mapbox [example](https://github.com/plotly/plotly.js/blob/07822743b03e9bc1a3decd429c02b0452af49732/src/plots/mapbox/index.js#L83-L90))
- call `restyle` with the values in that stash each in the modebar button handler (see [example](https://github.com/plotly/plotly.js/blob/07822743b03e9bc1a3decd429c02b0452af49732/src/components/modebar/buttons.js#L596-L614))

Reply:
@antoinerg let me know if you need help implementing this!

Reply:
> call restyle with the values in that stash each in the modebar button handler (see example)

I don't think I can reuse `resetView` because it does a `relayout` whereas we need a `restyle`.

Reply:
Right, no need to use `resetView()` found in `modebar/buttons.js`, but you should try to replicate the same pattern.

Reply:
Done in commit 506da4c! Let me know if it is satisfactory. 

I will now write Jasmine tests for the reset button. Is there anything else you would like me to add?

Reply:
> I will now write Jasmine tests for the reset button. Is there anything else you would like me to add?

Nop, once that jasmine test is in, this one will be good to go!

Reply:
Done in f8a4073